src/components/TagWall/TagWall.js

```diff
@@ -116,6 +118,7 @@ TagWall.propTypes = {
       id: string.isRequired,
       isSelected: bool,
       label: string.isRequired,
+      props: object,
```

Comment:
- Is `props` an ideal name for this? Or maybe `extra-props`? 🤔 
- We don't know the shape of `props` in this case, but it still may help to provide more info?

Reply:
🎉 Awesome stuff! A quick search of Carbon's repo wasn't able to produce any results, although I saw `attrs` used in places - Perhaps @joshblack could provide some guidance here? Otherwise, I'm good with props for now!

Reply:
Looking at this real quick, is this a forwarding-props use-case? 👀 In the code it seems like it's a part of the `items` array that is mapped over to return an `InteractiveTag`.

If so, I think this is a pattern that we've found ourselves doing for APIs like Downshift or DatePicker. Downshift might be the most comparable in that it has a similar need for items.

In general, I think we've realized that forwarding props in this way have either limited the types of changes that we can make, or puts us in a situation where it is hard to test whether a change breaks existing usage.

In the first case, the issue may manifest itself in props like `downshiftProps` where we forward things down to the Downshift library itself. This couples us to the version of downshift at a given point in time and has made it so that we can't upgrade without a major change. Similarly, if forwarding props to a component (especially in a component package future) we could run into the same situation.

In the latter case, having unbounded inputs has prevented us from identifying what we needed to test for while making updates. Our preference in this case would be to strictly type/define the interface for each prop so that we can enumerate the situations we have to test for and (hopefully) catch bugs in the future.

Similarly, in these cases we might try to invert the Component pattern:

```js
// From (pull syntax, receive items and decorate them)
<List items={items} />

// To (a) (implict method, add props with cloneElement or Context)
<List>
  {items.map(...)}
</List>

// To (b) (push method, give props and delegate to team to render)
<List renderItem={itemProps => ...} />

// To (c) (push method, hooks variant, return props based on options)
const [items] = useList(options);

<List>
  {items.map(item => ...)}
</List>
```

Reply:
@joshblack Thanks, that's very helpful! I think that's ideally where we want to go but it would constitute a breaking change